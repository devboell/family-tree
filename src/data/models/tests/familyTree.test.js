import knex from '../../connectors'
import FamilyTreeModel from '../familyTree'

describe('FamilyTreeModel', () => {
  let familyTreeTable
  let familyPartners

  beforeEach(async () => {
    await knex.migrate.rollback()
    await knex.migrate.latest()
    await knex.seed.run()
    familyTreeTable = await FamilyTreeModel.familyTreeTable(1)
    familyPartners = await FamilyTreeModel.familyPartners(familyTreeTable)
  })

  afterEach(async () => {
    await knex.migrate.rollback()
  })

  afterAll(() => {
    knex.destroy()
  })

  it('familyTreeTable', () => {
    expect(familyTreeTable).toMatchSnapshot()
  })

  it('familyPartners', () => {
    expect(familyPartners).toMatchSnapshot()
  })
})
