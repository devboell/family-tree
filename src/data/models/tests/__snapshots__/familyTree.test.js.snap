// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FamilyTreeModel familyPartners 1`] = `
Array [
  Object {
    "birthDate": 1031,
    "bornToId": null,
    "deathDate": 1083,
    "gender": "F",
    "id": 2,
    "isMonarch": 0,
    "name": "Matilde of Flanders",
    "title": "Queen of England",
  },
  Object {
    "birthDate": 1080,
    "bornToId": null,
    "deathDate": 1118,
    "gender": "F",
    "id": 5,
    "isMonarch": 0,
    "name": "Matilde of Scotland",
    "title": "Queen of England",
  },
  Object {
    "birthDate": 1113,
    "bornToId": null,
    "deathDate": 1151,
    "gender": "M",
    "id": 8,
    "isMonarch": 0,
    "name": "Geoffrey Plantagenet",
    "title": "Count of Anjou",
  },
  Object {
    "birthDate": 1122,
    "bornToId": null,
    "deathDate": 1204,
    "gender": "F",
    "id": 10,
    "isMonarch": 0,
    "name": "Eleanor of Aquitane",
    "title": "Queen of England",
  },
  Object {
    "birthDate": 1103,
    "bornToId": null,
    "deathDate": 1151,
    "gender": "F",
    "id": 14,
    "isMonarch": 0,
    "name": "Adeliza of Louvain",
    "title": "Queen of England",
  },
  Object {
    "birthDate": 1045,
    "bornToId": null,
    "deathDate": 1102,
    "gender": "M",
    "id": 16,
    "isMonarch": 0,
    "name": "Stephen II Henry",
    "title": "Count of Blois",
  },
]
`;

exports[`FamilyTreeModel familyTreeTable 1`] = `
Array [
  Object {
    "bornToId": null,
    "id": 1,
    "level": 0,
    "name": "William I",
    "partnerId": 2,
    "relationId": 1,
  },
  Object {
    "bornToId": 1,
    "id": 3,
    "level": 1,
    "name": "William Rufus",
    "partnerId": null,
    "relationId": null,
  },
  Object {
    "bornToId": 1,
    "id": 4,
    "level": 1,
    "name": "Henry I",
    "partnerId": 5,
    "relationId": 2,
  },
  Object {
    "bornToId": 1,
    "id": 4,
    "level": 1,
    "name": "Henry I",
    "partnerId": 14,
    "relationId": 5,
  },
  Object {
    "bornToId": 1,
    "id": 13,
    "level": 1,
    "name": "Robert Curthose",
    "partnerId": null,
    "relationId": null,
  },
  Object {
    "bornToId": 1,
    "id": 15,
    "level": 1,
    "name": "Adela of Normandy",
    "partnerId": 16,
    "relationId": 6,
  },
  Object {
    "bornToId": 2,
    "id": 6,
    "level": 2,
    "name": "William Adelin",
    "partnerId": null,
    "relationId": null,
  },
  Object {
    "bornToId": 2,
    "id": 7,
    "level": 2,
    "name": "Empress Matilde",
    "partnerId": 8,
    "relationId": 3,
  },
  Object {
    "bornToId": 6,
    "id": 17,
    "level": 2,
    "name": "Stephen of Blois",
    "partnerId": null,
    "relationId": null,
  },
  Object {
    "bornToId": 3,
    "id": 9,
    "level": 3,
    "name": "Henry II",
    "partnerId": 10,
    "relationId": 4,
  },
  Object {
    "bornToId": 4,
    "id": 11,
    "level": 4,
    "name": "Richard I",
    "partnerId": null,
    "relationId": null,
  },
  Object {
    "bornToId": 4,
    "id": 12,
    "level": 4,
    "name": "John I",
    "partnerId": null,
    "relationId": null,
  },
]
`;
